aws/resource_aws_spot_fleet_request_test.go

```diff
@@ -641,9 +641,10 @@ func TestAccAWSSpotFleetRequest_multipleInstanceTypesInSameAz(t *testing.T) {
 					testAccCheckAWSSpotFleetRequestExists(resourceName, &sfr),
 					resource.TestCheckResourceAttr(resourceName, "spot_request_state", "active"),
 					resource.TestCheckResourceAttr(resourceName, "launch_specification.#", "2"),
-					tfawsresource.TestCheckTypeSetElemNestedAttrs(resourceName, "launch_specification.*", map[string]string{
-						"instance_type":     "m1.small",
-						"availability_zone": "us-west-2a",
+					tfawsresource.TestMatchTypeSetElemNestedAttrs(resourceName, "launch_specification.*", map[string]*regexp.Regexp{
+						// match Regexp as instance_type is one of the t2.micro or t3.micro preferred types
+						"instance_type":     regexp.MustCompile(`t(2|3).micro`),
```

Comment:
Drive-by comment: You may find `tfawsresource.TestCheckTypeSetElemAttrPair()` easier to use in these cases, especially as we'll want to remove the hardcoded availability zones too 😄 

```go
					tfawsresource.TestCheckTypeSetElemAttrPair(resourceName, "launch_specification.*", "data.aws_ec2_instance_type_offering.available", "instance_type"),
```

Reply:
i thought so too! but i gave that one a try and from the output it doesn't seem to handle blocks like `launch_specification` ?  
```	
* Check 4/5 error: no TypeSet element "launch_specification.*", with value "t3.micro" 
```
even though you can see that value in the state that gets printed out :/

Reply:
@bflad, https://github.com/terraform-providers/terraform-provider-aws/pull/14843 could be an approach to supporting test checks with nested attributes like we have here.

Reply:
Oh yikes, yeah, `TestCheckTypeSetElemAttrPair()` is for attribute TypeSet and not block TypeSet. Sorry about that. I think this is okay for now and we can iterate on it later if necessary. 👍